core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -601,6 +616,7 @@ private long catchUp(JournalReader journalReader, long limit) {
             return journalReader.getNextSequenceNumber();
         }
       } catch (IOException e) {
+        // TODO(jiacheng): is it possible that an interrupt is initiated but still retrying here?
         LOG.warn("{}: Failed to read from journal: {}", mMaster.getName(), e);
         if (retry.attempt()) {
```

Comment:
@elega I vaguely recall you've seen problems where the master is shutting down but this keeps retrying. If that's the case (like we wrapped InterruptedException in IOException and that got caught here), we can improve that here.

Reply:
I looked at the logs. All attempts of journal flush retry finishes before the master starts to shutdown so there's no issue.